jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/StepsRunner.java

```diff
@@ -358,22 +358,27 @@ private void buildAndCacheApplicationLayers() {
   private void buildImage() {
     ProgressEventDispatcher.Factory childProgressDispatcherFactory =
         Verify.verifyNotNull(rootProgressDispatcher).newChildProducer();
-
-    results.builtImage =
+    results.builtImages =
         executorService.submit(
-            () ->
-                new BuildImageStep(
-                        buildContext,
-                        childProgressDispatcherFactory,
-                        results.baseImagesAndRegistryClient.get().images.get(0),
-                        realizeFutures(
-                            Verify.verifyNotNull(
-                                results
-                                    .baseImagesAndLayers
-                                    .get()
-                                    .get(results.baseImagesAndRegistryClient.get().images.get(0)))),
-                        realizeFutures(Verify.verifyNotNull(results.applicationLayers)))
-                    .call());
+            () -> {
+              List<Future<Image>> builtImages = new ArrayList<>();
+              for (Image image : results.baseImagesAndLayers.get().keySet()) {
+                Future<Image> builtImage =
+                    executorService.submit(
+                        () ->
+                            new BuildImageStep(
+                                    buildContext,
+                                    childProgressDispatcherFactory,
```

Comment:
Just realized this. I actually think this code (and the code in `obtainBaseImageLayers` that we updated) will break once we have multiple `Images` due to `childProgressDispatcherFactory` (the progress reporting thing), which I haven't explained. Will need to fix it when the time comes. Will be easy to fix, so not a concern at all.

Reply:
Sounds Good .I will be on the look out for the breaks.